packages/utils/src/time.ts

```diff
@@ -16,7 +16,7 @@ interface TimestampSource {
  * is more obvious to explain "why does my span have negative duration" than "why my spans have zero duration".
  */
 const dateTimestampSource: TimestampSource = {
-  nowSeconds: () => Date.now() / 1000,
+  nowSeconds: () => msToSec(Date.now()),
```

Comment:
We can prob just stop using the `nowSeconds` pattern as well, idk why it exists. A quick patch I have saved from before.

```diff
diff --git a/packages/utils/src/time.ts b/packages/utils/src/time.ts
index 932fffc9..ca59433a 100644
--- a/packages/utils/src/time.ts
+++ b/packages/utils/src/time.ts
@@ -1,24 +1,6 @@
 import { getGlobalObject } from './global';
 import { dynamicRequire, isNodeEnv } from './node';
 
-/**
- * An object that can return the current timestamp in seconds since the UNIX epoch.
- */
-interface TimestampSource {
-  nowSeconds(): number;
-}
-
-/**
- * A TimestampSource implementation for environments that do not support the Performance Web API natively.
- *
- * Note that this TimestampSource does not use a monotonic clock. A call to `nowSeconds` may return a timestamp earlier
- * than a previously returned value. We do not try to emulate a monotonic behavior in order to facilitate debugging. It
- * is more obvious to explain "why does my span have negative duration" than "why my spans have zero duration".
- */
-const dateTimestampSource: TimestampSource = {
-  nowSeconds: () => Date.now() / 1000,
-};
-
 /**
  * A partial definition of the [Performance Web API]{@link https://developer.mozilla.org/en-US/docs/Web/API/Performance}
  * for accessing a high resolution monotonic clock.
@@ -89,21 +71,14 @@ function getNodePerformance(): Performance | undefined {
 }
 
 /**
- * The Performance API implementation for the current platform, if available.
+ * Returns a timestamp in seconds since the UNIX epoch using the Date API.
  */
-const platformPerformance: Performance | undefined = isNodeEnv() ? getNodePerformance() : getBrowserPerformance();
-
-const timestampSource: TimestampSource =
-  platformPerformance === undefined
-    ? dateTimestampSource
-    : {
-        nowSeconds: () => (platformPerformance.timeOrigin + platformPerformance.now()) / 1000,
-      };
+export const dateTimestampInSeconds: () => number = () => Date.now() / 1000;
 
 /**
- * Returns a timestamp in seconds since the UNIX epoch using the Date API.
+ * The Performance API implementation for the current platform, if available.
  */
-export const dateTimestampInSeconds: () => number = dateTimestampSource.nowSeconds.bind(dateTimestampSource);
+const platformPerformance: Performance | undefined = isNodeEnv() ? getNodePerformance() : getBrowserPerformance();
 
 /**
  * Returns a timestamp in seconds since the UNIX epoch using either the Performance or Date APIs, depending on the
@@ -116,7 +91,10 @@ export const dateTimestampInSeconds: () => number = dateTimestampSource.nowSecon
  * skew can grow to arbitrary amounts like days, weeks or months.
  * See https://github.com/getsentry/sentry-javascript/issues/2590.
  */
-export const timestampInSeconds: () => number = timestampSource.nowSeconds.bind(timestampSource);
+export const timestampInSeconds =
+  platformPerformance === undefined
+    ? dateTimestampInSeconds
+    : () => (platformPerformance.timeOrigin + platformPerformance.now()) / 1000;
 
 // Re-exported with an old name for backwards-compatibility.
 export const timestampWithMs = timestampInSeconds;
```